### PR TITLE
Added behaviour endpoints

### PR DESCRIPTION
**Pull request by MinIcecream**
Created endpoints to modify/access the behaviour table. 
Enabled getting all, getting one, creating, updating, and deleting entries.
Manually tested operations using Postman.

### CODE REVIEW

**Review by aanxniee — COMMENTED** [review]
can you adding some ss to show your testing on postman

File: backend/typescript/models/behaviour.model.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Column, Model, Table } from "sequelize-typescript";
 
-@Table({ tableName: "behaviours" })
+@Table({ timestamps: false, tableName: "behaviours" })
```
Comment (aanxniee) [inline backend/typescript/models/behaviour.model.ts:3]: what does the timestamp do

**Review by aanxniee — COMMENTED** [review]
awesome work! seems right to me but let's add some ss so everyone reviewing can see it works. also did you make sure to test edge cases, i.e trying to delete something that doesn't exist, adding a behaviour name that's not valid

**Review by jerry-cheng5 — COMMENTED** [review]
Looks really solid, just a couple small comments for design choices

File: backend/typescript/services/interfaces/behaviourService.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+export interface BehaviourRequestDTO {
+  behaviourName: string;
+}
+
+export interface BehaviourResponseDTO {
+  id: string;
```
[the remark this answers is not in the document]
Comment (aanxniee) [reply backend/typescript/services/interfaces/behaviourService.ts:6]: agreed

File: backend/typescript/models/behaviour.model.ts (hunk 1 above)
Comment (MinIcecream) [reply backend/typescript/models/behaviour.model.ts:3]: Sequelize automatically makes like a createdAt column and stuff so this line just stops it from doing that

**Comment by MinIcecream** [comment]
1. Getting a specific behaviour, deleting a behaviour, and updating a behaviour now throws a 404 if the behaviour is not found.
2. Changed id field in the BehaviourResponseDTO to be a number.

3.  Testing:

  a. Getting all returns all behaviours in the database
image

  b. Getting a specific behaviour returns it if it exists
image

  c. Getting a specific behaviour returns 404 if it doesn't exist
image

  d. Creating a new behaviour returns the created behaviour and updates database
image

  e.  Updating an existing behaviour returns the updated behaviour and updates database
image

  f. Updating a nonexistent behaviour returns a 404 and doesn't affect database
image

g. Deleting an existing behaviour returns deleted behaviour's id and updates database
image

h. Deleting nonexistent behavour returns 404 and doesn't affect database 
image

**Comment by MinIcecream** [comment]
I also added the NotFoundError to the errorUtils file I found, but let me know if there's a better way to handle it.

File: backend/typescript/services/interfaces/behaviourService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (laks0407) [reply backend/typescript/services/interfaces/behaviourService.ts]: yeah me as well

File: backend/typescript/rest/behaviourRoutes.ts [hunk 3]
```diff
@@ -52,7 +52,11 @@ behaviourRouter.get("/:id", async (req, res) => {
     const behaviour = await behaviourService.getBehaviour(id);
     res.status(200).json(behaviour);
   } catch (e: unknown) {
-    res.status(500).send(getErrorMessage(e));
+    if (e instanceof NotFoundError) {
+      res.status(404).send(getErrorMessage(e));
+    } else {
+      res.status(500).send(getErrorMessage(e));
```
Comment (aanxniee) [inline backend/typescript/rest/behaviourRoutes.ts:58]: for 500 i dont think we want to send the error message back. 500 is internal error so ideally we want to log the error internally, confirm @jerry-cheng5

Comment (jerry-cheng5) [reply backend/typescript/rest/behaviourRoutes.ts:58]: Yea I agree, I think that's the proper way to handle 500 errors. 

(Normally its for security and shi and we don't reaaally have to worry about that, but its always good to practice the proper response conventions)

Comment (jerry-cheng5) [reply backend/typescript/rest/behaviourRoutes.ts:58]: So maybe just send some generic message back like:

An unexpected internal server error occurred. Please try again later.

or

An unexpected error occurred on our end. Please try again later.

**Review by aanxniee — APPROVED** [review]
lgtm, wait for @laks0407 @jerry-cheng5 to take a final look

**Review by jerry-cheng5 — APPROVED** [review]
Great work! Really impressed by how fast you did all this :) 

Everything looks good to me!

**Review by laks0407 — APPROVED** [review]
Looks great to me as well! Nice work :)
